internal/controller/http/v1/error.go

```diff
@@ -19,35 +20,95 @@ type response struct {
 
 func errorResponse(c *gin.Context, err error) {
 	var (
-		valErr          validator.ValidationErrors
+		validatorErr    validator.ValidationErrors
 		nfErr           sqldb.NotFoundError
-		nuErr           sqldb.NotUniqueError
+		notValidErr     dto.NotValidError
 		dbErr           sqldb.DatabaseError
+		NotUniqueErr    sqldb.NotUniqueError
 		amtErr          devices.AMTError
 		certExpErr      domains.CertExpirationError
 		certPasswordErr domains.CertPasswordError
 	)
 
 	switch {
-	case errors.As(err, &valErr):
-		c.AbortWithStatusJSON(http.StatusBadRequest, response{err.Error()})
+	case errors.As(err, &notValidErr):
+		notValidErrorHandle(c, notValidErr)
+
+		return
+	case errors.As(err, &validatorErr):
+		validatorErrorHandle(c, validatorErr)
+
+		return
+	case errors.As(err, &nfErr):
+		notFoundErrorHandle(c, nfErr)
+
+		return
+	case errors.As(err, &NotUniqueErr):
+		notUniqueErrorHandle(c, NotUniqueErr)
+
+		return
+	case errors.As(err, &dbErr):
+		dbErrorHandle(c, dbErr)
+
+		return
+	case errors.As(err, &amtErr):
+		amtErrorHandle(c, amtErr)
+
+		return
```

Comment:
why did a return become necessary for all these case statements?

Reply:
done